### PR TITLE
Fix to check chainerx device in ImageNet example

### PR DESCRIPTION
**Pull request by okapies**
Fix to check `device` is ChainerX device in the ImageNet example correctly. `device` is `chainer.Device` not `chainerx.Device`.

### CODE REVIEW

**Comment by niboshi** [comment]
Thanks!
`chainerx.is_available() and` can also be omitted.

File: examples/imagenet/train_imagenet.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def main():
     trainer.extend(extensions.Evaluator(val_iter, model, converter=converter,
                                         device=device), trigger=val_interval)
     # TODO(sonots): Temporarily disabled for chainerx. Fix it.
-    if not (chainerx.is_available() and isinstance(device, chainerx.Device)):
+    if not (chainerx.is_available() and device.xp is chainerx):
```
Comment (niboshi) [inline examples/imagenet/train_imagenet.py]:
```suggestion
    if device.xp is not chainerx:
```

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by okapies** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
LGTM

**Pull request merged**